var/spack/repos/builtin/packages/portage/package.py

```diff
@@ -34,23 +33,27 @@ class Portage(CMakePackage):
     depends_on("cmake@3.13:", type='build')
 
     depends_on('mpi', when='+mpi')
+    depends_on('kokkos', when='+kokkos')
+    depends_on('thrust', when='+thrust')
+    depends_on('jali', when='+jali')
 
     depends_on('tangram', when='+tangram')
-    depends_on('tangram+mpi', when='+tangram+mpi')
-    depends_on('tangram+jali', when='+tangram+jali')
-    depends_on('tangram+flecsisp', when='+tangram+flecsisp')
-    depends_on('tangram+thrust', when='+tangram+thrust')
-    depends_on('tangram+kokkos', when='+tangram+kokkos')
-    depends_on('tangram+cuda', when='+tangram+cuda')
+
+    tangram_variant = ['mpi', 'jali', 'openmp', 'thrust', 'kokkos', 'cuda']
+
+    # tangram depends on loop
+    for i in tangram_variant:
+        depends_on('tangram+' + i, when='+tangram+' + i)
+        depends_on('tangram~' + i, when='+tangram~' + i)
```

Comment:
- You should prefix class-level temporaries with `_` so they don't look like package attributes. (It's not going to really be a problem here since the `tangram_` prefix means it's not going to conflict with a spack attribute.)
- Use meaningful variable names; `i` is usually only appropriate for a simple linear numeric counter or matrix row
- The comment is superfluous: it's clearly a loop over dependencies

But since you don't reuse this list of variants, you should just simplify this to 
```suggestion
    for _variant in ['mpi', 'jali', 'openmp', 'thrust', 'kokkos', 'cuda']:
        depends_on('tangram+' + _variant, when='+tangram+' + _variant)
        depends_on('tangram~' + _variant, when='+tangram~' + _variant)
```
Note that loop variables end up in the same scope as the `for` loop, so use `_variant` instead of `variant`.